pkg/ebpf/tracee.go

```diff
@@ -1263,33 +1159,12 @@ func (t *Tracee) attachProbes() error {
 		}
 	}
 
-	// TODO: remove all tc programs (being attached to specific interfaces)
-
-	for _, iface := range t.netInfo.ifaces {
-		for _, tc := range []probes.Handle{
-			probes.DefaultTcIngress,
-			probes.DefaultTcEgress,
-		} {
-			err = t.probes.Attach(tc, iface)
-			if err != nil {
-				return err
-			}
-		}
-	}
-
 	return nil
 }
 
 func (t *Tracee) initBPF() error {
 	var err error
 
-	if t.netEnabled() {
-		// TODO: NET_ADMIN is needed by tcProbes, which are not declared as
-		// events so they don't have required capabilities. Once tcProbes gone
-		// we can remove this from the required capabilities set.
-		capabilities.GetInstance().Require(cap.NET_ADMIN)
```

Comment:
Isn't this required for cgroup programs as well?

Reply:
All the network-based events are dependent of a single "network base event":

![image](https://user-images.githubusercontent.com/7395852/212575752-d0c925e1-8962-475f-964d-0776e8642dd2.png)

and it sets the NET_ADMIN as a required capability.

The problem we had with tc hooks was that they were not used as "probe dependencies" for the network events so the logic of raising capabilities was not enough.

Now we can use the dependency like it should be used.